src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -454,22 +484,26 @@ def _record_compile_classpath(self, classpath, target, outdir):
       f.write(text)
 
   def _set_directory_digests_for_valid_target_classpath_directories(self, valid_targets, compile_contexts):
+    def _get_relative_classpath_for_target(target):
+      cc = self.select_runtime_context(compile_contexts[target])
+      if self.get_options().use_classpath_jars:
+        return fast_relpath(cc.jar_file.path, get_buildroot())
+      else:
+        return fast_relpath(cc.classes_dir.path, get_buildroot()) + '/**'
```

Comment:
```suggestion
        return '{}/**'.format(fast_relpath(cc.classes_dir.path, get_buildroot()))
```

Reply:
the irony of format is that it is seemingly more likely to result in getting `b'my/filename'/**`, since it isn't typesafe, heh.